### DISCUSSION TITLE
which setting to tweak for an "all included" html output

### DISCUSSION THREAD

**Discussion by horaciosamaniego** [marp-team/marp#516] [Q&A] [answered]
I am an avid Marp user through vscode. great tool!

how can we include (an equivalent of) the --self-contained  pandoc flag to get an standalone html output from  a .md in vscode.

I just couldn't find the right settings in my vscode, sorry.

I'm sure that this will help to spread the use this excellent tool.

cheers, 

H

**Comment by yhatt** [comment, marked as answer]
Marp official tools do not provide a way to bundle assets into a single HTML file, because it may be imperfect against references to assets that are dynamically generated in slides (e.g. generated by JavaScript): https://github.com/marp-team/marp-vscode/issues/142#issuecomment-668459792

Marp community recommends using **[monolith](https://github.com/Y2Z/monolith)**, a CLI tool to generate a standalone HTML file with assets.

```bash
monolith generated-by-marp.html -o self-contained.html
```

You also can convert Marp Markdown into the standalone HTML with one liner by using [Marp CLI](https://github.com/marp-team/marp-cli).

```
marp your-markdown.md -o - | monolith - -o output.html
```

### Related

- #132
- https://github.com/marp-team/marp/discussions/122#discussioncomment-775560
- https://github.com/marp-team/marp-cli/issues/159#issuecomment-884974386

**Reply by horaciosamaniego** [reply]
this is great info! Thanks so much for making this possible and being so helpful
cheers
